cleanup_job.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+import logging
+import requests
+from datetime import datetime, timedelta
+
+from cachito.workers.requests import get_requests_session
+from cachito.workers.config import get_worker_config
+from cachito.errors import CachitoError
+
+log = logging.getLogger(__name__)
+
+auth_session = get_requests_session(auth=True)
+session = get_requests_session()
+config = get_worker_config()
+
+
+def main():
+    """
+    Get all completed requests from the Cachito API.
+
+    :raise CachitoError: if the request to the Cachito API fails
+    """
+    # Get all completed requests
+    response = session.get(config.cachito_api_url + 'requests?state=complete')
+    if not response.ok:
+        raise CachitoError('Could not reach cachito API to get all completed requests')
+    json_response = response.json()
+    identify_stale_requests(json_response)
+
+
+def identify_stale_requests(requests_json):
+    """
+    Check if a completed Cachito request has reached end of life.
+
+    :param dict requests_json: json representation of all complete requests
+    """
+    stale_request_ids = []
+    current_time = datetime.now()
+    for request in requests_json['items']:
+        date_time_obj = datetime.strptime(request['updated'], '%Y-%m-%dT%H:%M:%S.%f')
+        if current_time - date_time_obj > timedelta(config.cachito_request_lifetime):
+            stale_request_ids.append(request['id'])
+
+    mark_as_stale(stale_request_ids)
+
+
+def mark_as_stale(stale_request_ids):
+    """
+    Mark the identified stale request ids as `stale` in cachito.
+
+    :param list stale_request_ids: list of request ids identified as stale
+    :raise CachitoError: if the request to the Cachito API fails
+    """
+    state = 'stale'
+    state_reason = 'The request has expired'
+    payload = {'state': state, 'state_reason': state_reason}
+    for request_id in stale_request_ids:
+        try:
+            log.info(f'Setting state of request {request_id} to `stale`')
+            request_rv = auth_session.patch(
+                f'{config.cachito_api_url.rstrip("/")}/requests/{request_id}',
+                json=payload, timeout=config.cachito_api_timeout)
+        except requests.RequestException:
+            msg = f'The connection failed when setting the `stale` state on request {request_id}'
+            log.exception(msg)
+            raise CachitoError(msg)
+
+        if not request_rv.ok:
+            log.error(
+                'The worker failed to set the `stale` state on request %d. The status was %d. '
```

Comment:
This won't be run by the `worker`, so perhaps this log message could remove the `The worker` section.